src/js/views/scoresheet.js

```diff
@@ -226,41 +215,38 @@ define('views/scoresheet',[
 
             //saves mission scoresheet
             $scope.save = function() {
-                if (!$scope.team || !$scope.stage || !$scope.round) {
+                if (!$scope.scoreEntry.team || !$scope.scoreEntry.stage || !$scope.scoreEntry.round) {
                     $window.alert('no team selected, do so first');
                     return $q.reject(new Error('no team selected, do so first'));
                 }
 
                 var data = angular.copy($scope.field);
-                data.uniqueId = $scope.uniqueId;
-                data.team = $scope.team;
-                data.stage = $scope.stage;
-                data.round = $scope.round;
-                // data.table = $scope.settings.table;
-                data.table = $scope.table;
+                data.scoreEntry = new $score($scope.scoreEntry);
+                data.team = $scope.scoreEntry.team;
+                data.stage = $scope.scoreEntry.stage;
+                data.round = $scope.scoreEntry.round;
+                data.table = $scope.scoreEntry.table;
                 data.referee = $scope.referee;
                 data.signature = $scope.signature;
-                data.score = $scope.score();
-
-                var fn = [
-                    'score',
-                    data.stage.id,
-                    'round' + data.round,
-                    'table' + data.table,
-                    'team' + data.team.number,
-                    data.uniqueId
-                ].join('_')+'.json';
+                data.scoreEntry.score = $scope.score();
+                data.scoreEntry.calcFilename();
```

Comment:
I'm not sure `calcFilename()` should be in `scoreEntry`: we're saving a scoresheet here, so theoretically, we'd also want to e.g. include the referee in the filename.

Also, the scoresheet is a whole different beast from a score entry (as you already noted yourself, because you did move e.g. stage, round, but not referee or signature).

The scoreEntry is not a 'subset' of the data of a scoresheet (thus not a simple property of it), but a really different thing. For example, it includes published status, whether it was editted, etc.
It *is* of course possible to create a new scoreEntry by filling some of its properties from a scoresheet.

Therefore, I'm not a big fan of changing the score property from a number to a scoreEntry object.
In fact, I would strongly suggest *reverting this part of the changes here*.

Then the server code must also be changed, by no longer picking the 'score' property out of the scoresheet object.
I consider this a good thing: the client should post the scoresheet, and can then separately post the score entry.
(Note that that is not as hard as it sounds, because if any of the two fail, both can be retried, even if the scoresheet post already worked.)

Reply:
I don't think it's hard change, but I do disagree with it.
The score entry may have some status property that are used in later steps of the score's life cycle, but it is being born out of a scoresheet.
Because of that, I tought it would be most efficient and logical to start the progress of the score entry with the scoresheet. The code here, is of course temporary (it really doesn't look that good, does it?) because The idea that came behind this, is to make the $score factory, and object the starts with the score from the moment it's created (as a scoresheet) to moment it's being saved in the server, and in later editing. After all, it is the same score from the beginning. 
Notice that it's not the server that extrudes them, but the $scores service, which will make it easier to change it later to a single full-process object

Reply:
Scoresheets are really just one of the ways in which a score can be born. Another is to manually enter them in the Scores overview.

Another use-case can be found in #261. When a scoresheet with a team signature is 'editted', a new scoresheet must be created. However, both scoresheets will refer to the same score entry. 

Also, any property of the score entry can be changed *independent* of the scoresheet, and when re-opening the scoresheet, that sheet must appear exactly as it was when the team signed it, even if it contained mistakes.

So although we can discuss about where to best 'convert' or 'sync' or instantiate one to the other, they are really different, and one is never just a subset of the other.

Reply:
I don't look at it a a subset, as I said, this is just temporary. (Iv'e got a little time now so I might be able to change that to show you what I mean, but it'll take me about a week). I'm saying that the scoresheet and the score are two states of the same entity. One is the state of creation, and the other is in the state of data manipulation.
I agree that the score can be created independently. That doesn't destroy the case that they are both the same entity, just in different forms.

The way I see it: 
The scoresheet is the mission points which the team singed off on. 
The score is the editable version of that.
I can see why that would seem like two different things, since they can change indepenetly (which make me wonder - why can they change independently? If the team singed on something, it should be permanent. It even says so in the game rules every year. If something like that is edited, it should be only on team aggrement - like you said in #261 it should be applied to all data, and it should be logged that it changed), but even if they do, that doesn't mean they don't represent the same score, just in different phases of its cycle.

Anyway, I'll try to show you what I mean by creating a score object that follows a score entry from the moment it's an empty scoresheet (or an entry of some other kind) to the moment it's edited, with full logs to show the changes that were applied to it.

Reply:
I thought about it, and I don't have the time to change it as I would want to, so I took it back to the way it was (Or the closest to it that I could)